router-bridge/js-src/plan.ts

```diff
@@ -281,7 +282,7 @@ export class BridgeQueryPlanner {
     let result = new Map<string, string>();
 
     subgraphs.names().forEach((name) => {
-      let sdl = printSchema(subgraphs.get(name).schema.toGraphQLJSSchema({}));
+      let sdl = printSchemaWithDirectives(subgraphs.get(name).schema);
```

Comment:
(This comment is more of an FYI.) Looking at [the PR](https://github.com/apollographql/federation-rs/pull/371) that introduced `subgraphs()`, it seems this method is used to generate subgraph schemas to validate subgraph responses against. Two things here:
1. I'm guessing here that the presence of directive applications in the subgraph schema shouldn't impede this subgraph response validation.
2. We're also presumably asserting here that the subgraph schemas with directive applications are valid GraphQL (which they ought to be), but I checked and that's not a new assertion in router (we use the default of `shouldValidate` when calling `Supergraph.build()` in router), so it should be fine (i.e., if there were bugs in how we generated/printed subgraph schemas, they wouldn't suddenly be triggered after this PR).

Reply:
> I'm guessing here that the presence of directive applications in the subgraph schema shouldn't impede this subgraph response validation.

It shouldn't affect that part of the code. One thing I'm noticing when running the router tests against this change is that it changes the query hash in a few tests because the hash will include directives if they exist. So, this will bust the cache, but that happens anyway when we bump the federation version. I think this will still be correct.